StatePrinter/OutputFormatters/JsonStyle.cs

```diff
@@ -118,104 +121,62 @@ string MakeString(List<Token> tokens, Dictionary<Reference, string> referencePat
         int MakeStringFromToken(List<Token> tokens, int pos, Dictionary<Reference, string> referencePaths, IndentingStringBuilder sb)
         {
             var token = tokens[pos];
-
+            TokenType? next = pos + 1 < tokens.Count ? tokens[pos + 1].Tokenkind : new TokenType?();
             int skip = 0;
 
-            string fieldnameColon = null;
             switch (token.Tokenkind)
             {
                 case TokenType.StartScope:
-                    sb.AppendFormatLine("{{");
-                    sb.Indent();
+                case TokenType.StartDict:
+                    if (next == TokenType.EndScope || next == TokenType.EndDict)
```

Comment:
in case of `(next == TokenType.EndScope || next == TokenType.EndDict)`  method `OptionalComma()` always return `""` - what are you trying to achieve with the `if`?